### PR TITLE
fix race condition in listener tests

### PR DESCRIPTION
**Pull request by aryszka**
fixing listener tests:
- wait for listener started on the happy path
- find random available port for test listeners

### CODE REVIEW

**Comment by aryszka** [comment]
this caused problems for me when ran the tests, can because of different platforms and timings etc

File: skipper_test.go [hunk 1]
```diff
@@ -34,74 +77,91 @@ func TestOptionsWithCertUsesHTTPS(t *testing.T) {
...
+	err = listenAndServe(proxy, &o)
 	if err == nil {
 		t.Fatal(err)
 	}
 }
 
 func TestWithWrongKeyPathFails(t *testing.T) {
-	o := Options{Address: ":9091",
+	a, err := findAddress()
+	if err != nil {
```
Comment (Raffo) [inline skipper_test.go]: This err check is empty, it must have the same behavior as above.

**Comment by Raffo** [comment]
You are totally right, it was pretty naive before and much better now. If you address my only comment it's OK to merge for me.

**Comment by aryszka** [comment]
thanks, fixed

**Pull request merged**